datastore/datastore.go

```diff
@@ -357,6 +357,54 @@ func checkMultiArg(v reflect.Value) (m multiArgType, elemType reflect.Type) {
 	return multiArgTypeInvalid, nil
 }
 
+// IgnoreFieldMismatch allows ignoring ErrFieldMismatch error while
+// reading or querying data.
+// WARNING: Ignoring ErrFieldMismatch can cause data loss
+func (c *Client) IgnoreFieldMismatch() {
```

Comment:
Consider expressing this as a client specific ClientIOption. Here is an example from storage: https://github.com/googleapis/google-cloud-go/blob/532a37c98680855fa38342bb7680062226594a0e/storage/option.go#L63-L75

Reply:
Updated the PR